pymeasure/instruments/toptica/ibeamsmart.py

```diff
@@ -22,99 +22,267 @@
+        if reply:
+            # anything else than '[OK]'.
+            self.adapter.connection.flush_read_buffer()
+            raise ValueError(
+                f"IBeamSmart: Error after command '{self.lastcommand}' with "
+                f"message '{reply}'")
 
     version = Instrument.measurement(
-           "ver", """ Firmware version number """,
+        "ver", """Get Firmware version number.""",
     )
 
     serial = Instrument.measurement(
-           "serial", """ Serial number of the laser system """,
+        "serial", """Get Serial number of the laser system.""",
     )
 
     temp = Instrument.measurement(
-            "sh temp",
-            """ temperature of the laser diode in degree centigrade.""",
+        "sh temp",
+        """Measure the temperature of the laser diode in degree centigrade.""",
     )
 
     system_temp = Instrument.measurement(
-            "sh temp sys",
-            """ base plate (heatsink) temperature in degree centigrade.""",
+        "sh temp sys",
+        """Measure base plate (heatsink) temperature in degree centigrade.""",
+    )
+
+    current = Instrument.measurement(
+        "sh cur",
+        """Measure the laser diode current in mA.""",
+        # TODO verify the unit
+    )
+
+    emission = Instrument.control(
+        "sta la", "la %s",
+        """Control emission status of the laser diode driver (bool).""",
+        validator=strict_discrete_set,
+        values=[True, False],
+        get_process=lambda s: True if s == 'ON' else False,
+        set_process=lambda v: "on" if v else "off",
+        check_set_errors=True,
     )
 
     laser_enabled = Instrument.control(
-            "sta la", "la %s",
-            """ Status of the laser diode driver.
-                This can be True if the laser is on or False otherwise""",
-            validator=strict_discrete_set,
-            values=[True, False],
-            get_process=lambda s: True if s == 'ON' else False,
-            set_process=lambda v: "on" if v else "off",
+        "sta la", "la %s",
+        """Control emission status of the laser diode driver (bool).
+
+        .. deprecated:: 0.12 Use attr:`emission` instead.
+        """,
+        validator=strict_discrete_set,
+        values=[True, False],
+        get_process=lambda s: True if s == 'ON' else False,
+        set_process=lambda v: "on" if v else "off",
+        check_set_errors=True,
     )
 
     channel1_enabled = Instrument.control(
-            "sta ch 1", "%s",
-            """ Status of Channel 1 of the laser.
-                This can be True if the laser is on or False otherwise""",
-            validator=strict_discrete_set,
-            values=[True, False],
-            get_process=lambda s: True if s == 'ON' else False,
-            set_process=lambda v: "en 1" if v else "di 1",
+        "sta ch 1", "%s",
+        """Control status of Channel 1 of the laser (bool).
+
+        .. deprecated:: 0.12 Use :attr:`ch_1.enabled` instead.
```

Comment:
Can you please add all the deprecations to a changelog entry?

@bmoneke I feel we should encourage this, otherwise it will be very hard to keep track of all freshly added deprecations when assembling the changelog come release time.

Reply:
Done.
I agree.